smqtk_relevancy/impls/libsvm_hik.py

```diff
@@ -211,12 +214,12 @@ def rank(self, pos, neg):
         #: :type: list[list]
         train_vectors = []
         num_pos = 0
-        for d in pos:
+        for desc_element in pos:
             train_labels.append(+1)
             # noinspection PyTypeChecker
-            train_vectors.append(d.vector().tolist())
+            train_vectors.append(desc_element.vector().tolist())  # type: ignore
```

Comment:
As usual, if we need to type-ignore on a random line, that means something else needs some more specificity maybe some local changes. This one seems to be related to the overloaded use of `pos` as input and the set-cast of the input, so mypy get's confused. Also, `vector()` has an optional return. This might want to have a way to handle when `vector()` returns None instead of just raising an AttributeError during runtime.

Reply:
Instead of commenting on all the other inline type-ignores, given them a second pass and see *why* you have to pass them an ignore for mypy's warnings, or if the warning is actually legitimate and motivates a revision.